onadata/apps/api/permissions.py

```diff
@@ -277,6 +277,9 @@ def has_object_permission(self, request, view, obj):
                 isinstance(obj.content_object, DataView)):
             return False
 
+        if 'key' in request.query_params or view.action == 'partial_update':
```

Comment:
how does this work? is it still validating the user has permissions on the form this widget is attached to?